### ISSUE TITLE
Can't build an exonum-0.10.3 project on Windows, missing libsodium binaries

### ISSUE THREAD

**Issue by simplesmiler** [exonum/exonum#1449]
From what I could gather, `exonum_libsodium-sys-0.0.20` on Windows depends on prebuilt binaries of `libsodium-1.0.16`, but for whatever reason they are [not available from the download page](https://download.libsodium.org/libsodium/releases/) anymore.

<details>
<summary>Complete log:</summary>

```
> C:/Users/simple/.cargo/bin/cargo.exe build --color=always --all --all-targets
   
   Compiling exonum_libsodium-sys v0.0.20
   Compiling actix-web v0.7.18
   Compiling reqwest v0.9.10

error: failed to run custom build command for `exonum_libsodium-sys v0.0.20`

Caused by:
  process didn't exit successfully: `C:\<project_dir>\target\debug\build\exonum_libsodium-sys-0464a1ecca4cf06a\build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at '
"powershell" "-Command" "([Net.ServicePointManager]::SecurityProtocol = \'Tls12\') -and ((New-Object System.Net.WebClient).DownloadFile(\"https://download.libsodium.org/libsodium/releases/libsodium-1.0.16-msvc.zip\", \"C:\\<project_dir>\\target\\debug\\build\\exonum_libsodium-sys-5fad3acf07c31fff\\out/installed/libsodium-1.0.16-msvc.zip\"))"

Exception calling "DownloadFile" with "2" argument(s): "The remote server returned an error: (404) Not Found."
At line:1 char:2
+ ([Net.ServicePointManager]::SecurityProtocol = 'Tls12') -and ((New-Ob ...
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException
 
', C:\Users\simple\.cargo\registry\src\github.com-1ecc6299db9ec823\exonum_libsodium-sys-0.0.20\build.rs:340:9
stack backtrace:
   0: backtrace::backtrace::trace_unsynchronized
             at C:\Users\VssAdministrator\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.29\src\backtrace\mod.rs:66
   1: std::sys_common::backtrace::_print
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\sys_common\backtrace.rs:47
   2: std::sys_common::backtrace::print
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\sys_common\backtrace.rs:36
   3: std::panicking::default_hook::{{closure}}
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:200
   4: std::panicking::default_hook
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:214
   5: std::panicking::rust_panic_with_hook
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:477
   6: std::panicking::continue_panic_fmt
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:384
   7: std::panicking::begin_panic_fmt
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:339
   8: build_script_build::download_compressed_file
             at .\build.rs:340
   9: build_script_build::main
             at .\build.rs:364
  10: std::rt::lang_start::{{closure}}<()>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\src\libstd\rt.rs:64
  11: std::rt::lang_start_internal::{{closure}}
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\rt.rs:49
  12: std::panicking::try::do_call<closure,i32>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:296
  13: panic_unwind::__rust_maybe_catch_panic
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libpanic_unwind\lib.rs:82
  14: std::panicking::try
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:275
  15: std::panic::catch_unwind
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panic.rs:394
  16: std::rt::lang_start_internal
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\rt.rs:48
  17: std::rt::lang_start<()>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\src\libstd\rt.rs:64
  18: main
  19: invoke_main
             at d:\agent\_work\3\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  20: __scrt_common_main_seh
             at d:\agent\_work\3\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  21: BaseThreadInitThunk
  22: RtlUserThreadStart

Process finished with exit code 101
```

</details>

I've looked into a couple of ways to workaround the issue, but the only reasonable way seems to be upgrading to `exonum-0.11.0`, which depends on `libsodium-1.0.17` binaries, which are still available. But assuming that they can disappear again, I hope that there is a better solution.

The main culprit seems to be that the `0.0.x` semver of `exonum_libsodium-sys` is very "restrictive" (not allowing to use `exonum_libsodium-sys-0.0.21` along with `exonum-0.10.3`).

**Comment by aleksuss** [comment]
Hi. Exonum 0.10.4 has been released which contains `exonum_sodiumoxide-0.0.23` and `exonum_libsodium-sys-0.0.23` correspondingly. Now it should compile without any problems.

**Comment by simplesmiler** [comment]
Thanks for releasing 0.10.4! It solved the issue, at least for now.

I guess removal of libsodium binaries is a rare occasion, so it seems that there's no further action that needs to be taken.
